File: output/ChatGPT/codenet/C++/Java/s353226283.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Scanner;

public class s353226283 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        ArrayList<Integer> vec = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            int t = sc.nextInt();
            vec.add(t);
        }
        Collections.sort(vec);
        int ans = Integer.MAX_VALUE;
        for (int i = 0; i + k - 1 < n; i++) {
            int diff = vec.get(i + k - 1) - vec.get(i);
            if (diff < ans) {
                ans = diff;
            }
        }
        System.out.println(ans);
    }
}

File: output/ChatGPT/codenet/C++/Python/s353226283.py
n, k = map(int, input().split())
vec = list(map(int, input().split()))
vec.sort()
ans = float('inf')
for i in range(n - k + 1):
    diff = vec[i + k - 1] - vec[i]
    if diff < ans:
        ans = diff
print(ans)